src/sync/incomingSyncUtils.js

```diff
@@ -744,3 +757,138 @@ export const integrateRecord = (database, settings, syncRecord) => {
     // Handle unexpected |changeType|.
   }
 };
+
+// Translations for merge logic.
+// TODO: bind translations to |DataType| constants to avoid breakage on schema update.
+const RECORD_TYPE_TO_TABLE = {
+  Item: {
+    StocktakeItem: {
+      field: 'item',
+    },
+    TransactionItem: {
+      field: 'item',
+    },
+    ItemBatch: {
+      field: 'item',
+    },
+    RequisitionItem: {
+      field: 'item',
+    },
+  },
+  Name: {
+    ItemBatch: {
+      field: 'supplier',
+    },
+    Transaction: {
+      field: 'otherParty',
+      setterMethod: 'setOtherParty',
+    },
+    Requisition: {
+      field: 'otherStoreName',
+    },
+  },
+};
+
+const RECORD_TYPE_TO_MASTERLIST = {
+  Item: {
+    MasterListItem: {
+      field: 'item',
+    },
+  },
+  Name: {
+    MasterListNameJoin: {
+      field: 'name',
+    },
+  },
+};
+
+/**
+ * Merge two existing records. One record is retained and the other is merged. After the objects
+ * are merged, the merged object is deleted from the database. The merge operation updates the
+ * fields of the retained record to reference the same data as the merged object.
+ *
+ * @param  {Realm}   database            Local database.
+ * @param  {Object}  settings            Local settings.
+ * @param  {string}  internalRecordType  Internal record type for merge operation.
+ * @param  {Object}  syncRecord          Data representing the sync record.
+ */
+export const mergeRecords = (database, settings, internalRecordType, syncRecord) => {
```

Comment:
I think this guy is probably the next on the chopping block 😆!

Reply:
maybe, but I can't see a good separation of the logic where it would make it easier than just having it all sequential, a part from the switch at the end. Sometimes a method just gotta be big 🤷‍